classes/05class/make/README.md

```diff
@@ -0,0 +1,185 @@
+
+***Contents***
+- [What is Make](#what-is-make)
+- [Why use Make?](#why-use-make)
+- [Makefile](#makefile)
+- [Running some code](#running-some-code)
+    - [Multiple Targets](#multiple-targets)
+    - [Pre-Requisites](#pre-requisites)
+    - [Variables](#variables)
+- [Mixing together](#mixing-together)
+
+## What is Make?
+Make is a tool to execute a sequence of commands (like a recipe) based on targets defined in a specific file.
+Commonly used to compile and build applications from its source code, the make utility has became very popular in the DevOps community, being used to create scripts for automation.
+
+## Why use Make?
+One of the biggest benefits of the Make utility is the ability to run in multiple platforms and shells. For example, if an engineer creates automations using bash scripts, another engineer using a different shell, like zsh, might have problems executing those scripts locally.
+
+By using make, the same script can be used in any Linux and MacOS operating system as long as the Make utility is installed. This approach improved the collaboration in DevOps/Cloud teams and made Make a very popular tool in the community.
+
+## Makefile
+The Makefile contains the targets and instructions that will be used when the Make utility is executed. It can contain variables, conditions (`if` and `else`) and loops (like `for`).
+
+As you can see the Makefile can be very complex and it can be used for multiple purposes. For this course and the pupose of what we'll be mostly using, we'll only cover the basics of it.
+
+## Running some Code
+Let's get our hands into the Make utility and see how it works.
+
+The Make file is, basically, made of targets and instructions, and the syntax follow the below:
+
+```Makefile
+target: pre-requisites
+    instructions
+```
+
+For the famous Hello World, we would have something like the code below to print `Hello World` in the console:
+
+```Makefile
+say_helloworld:
+    @echo "Hello World"
+```
+By executing the command `make say_helloworld` or even just `make` without any target in the directory with a Makefile with the content above, we would get the following output:
+
+```shell
+$ make say_helloworld
+Hello World
+$
+$ make 
+Hello World
+$
+```
+In this case, the target is `say_helloworld` and the instruction is the `@echo "Hello World"`. The target can be a binary file or just the name for the instructions, like our example above. In this case, those targets are called **phony targets**. From now on, we'll consider all our targets as phony target and we'll include the `.PHONY` directive in our Makefile, as below:
+
+```Makefile
+.PHONY: say_helloworld
+say_helloworld:
+    @echo "Hello World"
+```
+For more details of the reason behind that, you can check [this link](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)
+
+Because there is a single target, if the target is not specified, the single target is execued anyway.
```

Comment:
```suggestion
Because there is a single target, if the target is not specified, the first target of the `Makefile` will be executed.

There is a convention to create the first target as `all`, so when running just `make` it will invoke several targets:

```Makefile
all: target1 target2 target3
```